build.gradle.kts

```
@@ -0,0 +1,71 @@
+import com.example.buildsrc.MyExtension
+
+plugins {
+    id("java")
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnitPlatform()
+}
+
+allprojects{
+    extensions.create("Extension", MyExtension::class.java)
+
+    val addElementNoProjectAllScope by extra { _: Unit ->
+        configure<MyExtension> {
+            contents.add("addElementNoProjectAllScope")
+        }
+    }
+
+    //passing a no-argument lambda makes the val initialized with lambda result
+    val addElementFakeLambdaAllScope by extra {
+        configure<MyExtension> {
+            contents.add("addElementFakeLambdaAllScope")
+        }
+    }
+}
+
+val addElementNoProjectExtScope by extra { _: Unit ->
```

Can it be just
```suggestion
val addElementNoProjectExtScope by extra { () ->
```
i.e. empty list of params? or is this single throw-away `Unit` param necessary?

---

The syntax you proposed throws this error:

Not enough information to infer type variable T
Property delegate must have a 'getValue(Build_gradle, KProperty*>)' method. None of the following functions are suitable.

- getValue(Any?, KProperty<*>) defined in org.gradle.kotlin.dsl.InitialValueExtraPropertyDelegate